packages/types/mongoose-store/index.ts

```diff
@@ -0,0 +1,64 @@
+import * as Promise from 'bluebird';
+import * as mongoose from 'mongoose';
+
+export default class MongooseStore<T>{
+  constructor(protected conn: mongoose.Connection) {
+
+  }
+}
+
+type ID = string | number;
+
+export interface Store<T> {
+  query(): ExecutableChainQuery<T>;
+  or(): ChainQuery<T>;
+  find(query: QueryParams<T>): Promise<T[]>;
+  create(data: T): Promise<T>;
+  update(id: ID, data: T): Promise<T>;
+  delete(id: ID): Promise<T>;
+}
+
+interface Predicate<T> {
```

Comment:
An alternative to having a separate storage module might be having a Repository for each data-accessing module and exported from them, which we could freely implement in straight-up mongoose.

For example we'd have `@raincatcher/workorder/Repository#getUnfinishedWorkordersForUser(id)`.